### PR TITLE
406831:  Open AI role for service mi

### PR DESCRIPTION
**Pull request by rajeshkaremane**
# **What this PR does / why we need it**:
*A brief description of changes being made*
*Link to the relevant work items: *
[AB#406831](https://dev.azure.com/defragovuk/93c65135-d16b-49d6-a191-4a5313532779/_workitems/edit/406831)
Created script to add mi object to access group

# **Special notes for your reviewer**
*Any specific actions or notes on review?*

# Testing
*Any relevant testing information and pipeline runs.*
https://dev.azure.com/defragovuk/DEFRA-FFC/_build/results?buildId=619203&view=results
https://dev.azure.com/defragovuk/DEFRA-FFC/_build/results?buildId=620959&view=results
https://dev.azure.com/defragovuk/DEFRA-FFC/_build/results?buildId=619208&view=results
https://dev.azure.com/defragovuk/DEFRA-FFC/_build/results?buildId=620949&view=results
https://dev.azure.com/defragovuk/DEFRA-FFC/_build/results?buildId=620918&view=results
image

# Checklist (please delete before completing or setting auto-complete)
- [x] Story Work items associated (not Tasks)
- [x] Successful testing run(s) link provided
- [x] Title pattern should be `{work item number}: {title}`
- [x] Description covers all the changes in the PR
- [ ] This PR contains documentation
- [ ] This PR contains tests

# **How does this PR make you feel**:
gif

### CODE REVIEW

File: infra/services/coreai/deploy-services-infra.yaml [hunk 1]
```diff
@@ -40,7 +40,8 @@ parameters:
     type: string
     values:
       - application-domain
-      - application-registration      
+      - application-registration
+      - application-rbac
```
Comment (kbabigumira) [inline infra/services/coreai/deploy-services-infra.yaml:44]: If this is specific for Open AI, should the name be more specific e.g. application-openai-rbac?

Alternatively, we made the variable names more generic and not include openai

Comment (rajeshkaremane) [reply infra/services/coreai/deploy-services-infra.yaml]: Fixed. Also allowed option to pass an array

File: infra/services/coreai/coreai-mcu-generation-oai/env-vars/dev1.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+variables:
+  accessGroupMiList: '[{"groupName": "$(openAiUserGroup)",  "miSuffixList": "coreai-mcu-generation-oai"}]'
```
Comment (DaRoszko) [inline infra/services/coreai/coreai-mcu-generation-oai/env-vars/dev1.yaml:2]: if these are all the same.. just var overrides.. maybe better in common?

Comment (rajeshkaremane) [reply infra/services/coreai/coreai-mcu-generation-oai/env-vars/dev1.yaml]: moved to common variable file

**Review by ppotturi — APPROVED** [review]

**Review by ajaysoni-cap — APPROVED** [review]

**Pull request merged**